src/use-stack.ts

```diff
@@ -0,0 +1,45 @@
+import { useState, useMemo } from 'react';
+
+export const useStack = <T>(initialState: T[] = []) => {
+  const [value, setValue] = useState(initialState);
+
+  const handlers = useMemo(
+    () =>
+      Object.assign(value, {
```

Comment:
Is this necessary? I mean, you can omit this `Object.assign` since you're not really using the `value` object here. Just returning the handlers from the hook should be enough.

Reply:
Done ✔️